server/src/test/scala/com/pennsieve/discover/handlers/SyncHandlerSpec.scala

```diff
@@ -140,5 +132,68 @@ class SyncHandlerSpec
           )
         )
     }
+
+    "skip Athena results with datasetID == 0 and set real version for a result with version == 0" in {
+      val ds1 = TestUtilities.createDatasetV1(ports.db)()
+      val ds1V2 = TestUtilities.createNewDatasetVersion(ports.db)(ds1.datasetId)
+
+      val req1 = DatasetDownload(
+        ds1.datasetId,
+        ds1.version,
+        Some(DownloadOrigin.AWSRequesterPayer),
+        Some("REQID1"),
+        OffsetDateTime.of(2020, 11, 10, 10, 3, 1, 0, ZoneOffset.UTC)
+      )
+      val req2 = DatasetDownload(
+        ds1.datasetId,
+        0, // this should be replaced by ds1V2.version when it is inserted into Postgres
+        Some(DownloadOrigin.AWSRequesterPayer),
+        Some("REQID2"),
+        OffsetDateTime.of(2020, 11, 10, 10, 3, 6, 0, ZoneOffset.UTC)
+      )
+      val req3 = DatasetDownload(
+        0, // this 0 means that it should be ignored since it cannot be inserted into Postgres (violates foreign key constraint)
+        1,
+        Some(DownloadOrigin.AWSRequesterPayer),
+        Some("REQID3"),
+        OffsetDateTime.of(2020, 11, 10, 10, 3, 12, 0, ZoneOffset.UTC)
+      )
+      ports.athenaClient
+        .asInstanceOf[MockAthenaClient]
+        .datasetDownloadsForRange = Some(List(req1, req2, req3))
+
+      syncClient
+        .syncAthenaDownloads(
+          LocalDate.of(2020, 11, 10),
+          LocalDate.of(2020, 11, 11)
+        )
+        .await
+
+      val responseMetrics = TestUtilities.getDatasetDownloads(ports.db)(
+        LocalDate.of(2020, 11, 10),
+        LocalDate.of(2020, 11, 11)
+      )
+
+      responseMetrics shouldBe
+        List(
+          req1.copy(downloadedAt = convertToLocalOffset(req1.downloadedAt)),
+          req2.copy(
+            version = ds1V2.version,
+            downloadedAt = convertToLocalOffset(req2.downloadedAt)
+          )
+        )
+    }
+  }
+
+  // I believe the DB connection returns the downloadedAt values in DatasetDownloads in the local time offset.
+  // Without the conversion to the local offset tests were failing locally on developer machines,
+  // but passing in CI where the system timezone is UTC.
```

Comment:
fwiw another thing that could be happening, is that Postgres stores timestamps in microseconds, but the the java time classes like OffsetDateTime will use the system's precision, which can either be nanoseconds or microseconds. So when its in nanoseconds equality comparison fails in unit tests. And its also why it tends to fail locally (esp on newer macbooks) and pass in CI. Or at least that's what I've been seeing recently.

Reply:
When I run the tests on my laptop without using the `convertToLocalOffset()` function the results differ exactly by the offset that would be in effect on that date. At least by the time they are `OffsetDateTime` instances.

```scala
Expected :List(DatasetDownload(1,1,Some(AWSRequesterPayer),Some(REQID1),2020-11-10T10:03:01Z), DatasetDownload(1,2,Some(AWSRequesterPayer),Some(REQID2),2020-11-10T10:03:06Z))
Actual   :List(DatasetDownload(1,1,Some(AWSRequesterPayer),Some(REQID1),2020-11-10T05:03:01-05:00), DatasetDownload(1,2,Some(AWSRequesterPayer),Some(REQID2),2020-11-10T05:03:06-05:00))
```

Reply:
ah gotcha! good to know!